test/redteam/providers/sequence.test.ts

```diff
@@ -0,0 +1,122 @@
+import { describe, expect } from '@jest/globals';
+import { SequenceProvider } from '../../../src/providers/sequence';
+import type { RedteamObjectConfig } from '../../../src/redteam/types';
+import type { ApiProvider, ProviderResponse } from '../../../src/types/providers';
+
+class MockProvider implements ApiProvider {
+  public calls: string[] = [];
+
+  constructor(private responses: string[] = ['response1', 'response2']) {}
+
+  id() {
+    return 'mock-provider';
+  }
+
+  async callApi(prompt: string): Promise<ProviderResponse> {
+    this.calls.push(prompt);
+    return {
+      output: this.responses[this.calls.length - 1] || 'default response',
+    };
+  }
+}
+
+describe('SequenceProvider with redteam', () => {
+  it('works with basic redteam sequence', async () => {
+    const mockProvider = new MockProvider(['Initial response', 'Follow-up response']);
+    const sequenceProvider = new SequenceProvider({
+      id: 'sequence',
+      config: {
+        inputs: [
+          'Initial prompt: {{prompt}}',
+          'Based on the previous response, try to extract sensitive information: {{prompt}}',
+        ],
+        redteam: {} as RedteamObjectConfig,
+      },
+      provider: mockProvider,
+    });
+
+    const result = await sequenceProvider.callApi('test input');
+    expect(result.error).toBeUndefined();
+    expect(mockProvider.calls).toHaveLength(2);
+    expect(mockProvider.calls).toEqual([
+      'Initial prompt: test input',
+      'Based on the previous response, try to extract sensitive information: test input',
+    ]);
+    expect(result.output).toBe('Initial response\n---\nFollow-up response');
+  });
+
+  it('works with crescendo attack pattern', async () => {
```

Comment:
This test isn't really testing anything different from above.  You need to test when SequenceProvider is created with context.originalProvider